theano/tensor/elemwise.py

```diff
@@ -1037,12 +1037,26 @@ def _c_all(self, node, nodename, inames, onames, sub):
         # the index of the last of these aliased outputs.
 
         # We generate the C code of the inner loop using the scalar op
+        if self.openmp:
+            # If we are using openmp, we need to get rid of the "goto"
+            # statement in sub['fail']. For now we recreate it here.
+            fail = '''
+                {
+                    %(failure_var)s = %(id)s;
+                    if (!PyErr_Occurred()) {
+                        PyErr_SetString(PyExc_RuntimeError,
+                            "Unexpected error in an Op's C code. "
+                            "No Python exception was set.");
+                    }
+                }''' % sub
```

Comment:
I missed that part. I think it will fix the openmp part. Instead of duplicating this code and we will forget it when we will update the master version of it, what about calling gof.cc.failure_code(sub, use_goto=False)? Add to it a use_goto=True param that allow to remove the goto.

With that small changes, this PR does in part what I suggested.